### PR TITLE
cilium: split off handling of lb svc type in datapath

### PR DESCRIPTION
**Pull request by borkmann**
See commit msgs.

### CODE REVIEW

File: pkg/loadbalancer/loadbalancer.go [hunk 1]
```diff
@@ -71,10 +72,12 @@ func CreateSvcFlag(svcLocal, sessionAffinity bool, svcTypes ...SVCType) ServiceF
 		switch svcType {
 		case SVCTypeExternalIPs:
 			flags |= serviceFlagExternalIPs
-		case SVCTypeHostPort:
-			flags |= serviceFlagHostPort
 		case SVCTypeNodePort:
 			flags |= serviceFlagNodePort
+		case SVCTypeLoadBalancer:
```
Comment (tklauser) [inline pkg/loadbalancer/loadbalancer.go:77]: Might be worth to expand `TestCreateSvcFlag` to cover `SVCTypeLoadBalancer`?

File: pkg/loadbalancer/loadbalancer.go [hunk 2]
```diff
@@ -120,8 +125,8 @@ func (s ServiceFlags) SVCTrafficPolicy() SVCTrafficPolicy {
 func (s ServiceFlags) String() string {
 	var strTypes []string
 	typeSet := false
-	sType := s & (serviceFlagExternalIPs | serviceFlagHostPort | serviceFlagNodePort)
-	for _, svcType := range []SVCType{SVCTypeExternalIPs, SVCTypeHostPort, SVCTypeNodePort} {
+	sType := s & (serviceFlagExternalIPs | serviceFlagHostPort | serviceFlagNodePort | serviceFlagLoadBalancer)
```
Comment (tklauser) [inline pkg/loadbalancer/loadbalancer.go:128]: Same here, maybe extend `TestServiceFlags_String` to cover `SVCTypeLoadBalancer`?

**Review by tklauser — APPROVED** [review]
LGTM. Two small remarks regarding extending the unit tests. Probably not worth respinning just for these.

**Review by aanm — CHANGES_REQUESTED** [review]
LGTM but needs e2e that we can run on GKE since it offers a LB

**Comment by borkmann** [comment]
test-me-please

**Comment by borkmann** [comment]
(all tests green)

**Comment by borkmann** [comment]
test-me-please

**Comment by borkmann** [comment]
test-gke

**Comment by borkmann** [comment]
test-focus K8sService*

**Review by nebril — APPROVED** [review]

**Comment by borkmann** [comment]
(2 GKE test flakes, unrelated: https://jenkins.cilium.io/job/Cilium-PR-K8s-GKE/1717/)

**Review by aanm — APPROVED** [review]

**Pull request merged**